src/app/core/authentication/authentication.service.ts

```diff
@@ -89,11 +89,17 @@ export class AuthenticationService {
     this.storage = this.rememberMe ? localStorage : sessionStorage;
 
     if (environment.oauth.enabled) {
-      let httpParams = new HttpParams();
-      httpParams = httpParams.set('client_id', 'community-app');
-      httpParams = httpParams.set('grant_type', 'password');
-      httpParams = httpParams.set('client_secret', '123');
-      return this.http.disableApiPrefix().post(`${environment.oauth.serverUrl}/oauth/token`, {}, { params: httpParams })
+      let bodyData = new HttpParams();
+      bodyData = bodyData.set('client_id', 'community-app');
+      bodyData = bodyData.set('grant_type', 'password');
+      bodyData = bodyData.set('client_secret', '123');
```

Comment:
I think this client_secret needs to be encrypted. Should we ? @karantakalkar

Reply:
I am not completely sure how to go forward with this. Didn't test it locally. But these are consumer secrets, how would encoding help over here. This is an https connection so there won't be issues with snipping or man in the middle attacks.